src/sonic_ax_impl/mibs/vendor/cisco/ciscoEntityFRUControlMIB.py

```diff
@@ -0,0 +1,106 @@
+import subprocess
+import re
+
+from sonic_ax_impl import mibs
+from ax_interface import MIBMeta, ValueType, MIBUpdater, MIBEntry, SubtreeMIBEntry
+from ax_interface.encodings import ObjectIdentifier
+
+import pdb
+
+class PowerStatusHandler:
+    """
+    Class to handle the SNMP request
+    """
+    def __init__(self):
+        """
+        init the handler
+        """
+
+        # Holds the number of supported PSU
+        self.num_psus = self._getNumOfSupportedPsu()
+
+    def _getNumOfSupportedPsu(self):
+        """
+        :return: the number of supported PSU
+        """
+        try:
+            output = subprocess.check_output(['psuutil', 'numpsus'])
+            if str(output[0]).isdigit():
+                return int(output)
+        except subprocess.CalledProcessError:
+            mibs.logger.error("failed to run psuutil tool")
+        except IndexError:
+            mibs.logger.error("failed to get he number of supported PSU")
+
+        return 0
+
+    def _getPsuIndex(self, sub_id):
+        """
+        Get the PSU index from sub_id
+        :return: the index of supported PSU
+        """
+        if not sub_id or len(sub_id) > 1:
+            return None
+
+        psu_index = int(sub_id[0])
+
+        if psu_index < 1 or psu_index > self.num_psus:
+            return None
+
+        return psu_index
+
+    def get_next(self, sub_id):
```

Comment:
>get_next [](start = 8, length = 8)

Discussed with @oleksandrivantsiv the same thing before.

I am not sure you are following the interface design of other SubtreeMIBEntry updater. This is an example [https://github.com/Azure/sonic-snmpagent/blob/master/src/sonic_ax_impl/mibs/ietf/rfc4363.py](https://github.com/Azure/sonic-snmpagent/blob/master/src/sonic_ax_impl/mibs/ietf/rfc4363.py).

update_data() will fill a map and a sorted list
get_next() will bisect_right in the sorted list. Keep it as easy as possible
fdb_ifindex() is an index function. It just index into the map. Keep it as easy as possible

Reply:
The PSU MIB does not use updater, it gets the PSU status directly  from plugin without caching. 
Also it supports the querying the PSU subtree.

Reply:
You are expected to follow the SubtreeMIBEntry design. It is weird to check PSU status for each snmp query.

Reply:
I see your point. I am ok with "read the status on demand only".
One design principle of SubtreeMIBEntry is to make updater class not to worry about sub_id/OID calculations, except the function get_next(), which simply get next sub_id based on a sorted array. All the dirty calculation about sub_id/OID should be already handled by SubtreeMIBEntry itself.

Could you follow this principle?

Reply:
From my point of view, the current implementation follows such principle because PSU MIB does not has updater at all. It consists the SubtreeMIBEntry implementation.